### PR TITLE
Add events on schain pause and resume

### PR DESCRIPTION
**Pull request by DimaStebaev**

### CODE REVIEW

**Review by yavrsky — APPROVED** [review]

File: proxy/contracts/mainnet/MessageProxyForMainnet.sol [hunk 1]
```diff
@@ -100,6 +100,20 @@ contract MessageProxyForMainnet is SkaleManagerClient, MessageProxy, IMessagePro
         uint256 newValue
     );
 
+    /**
+     * @dev Emitted when the schain is paused
+     */
+    event SchainPaused(
+        bytes32 schainHash
```
Comment (payvint) [inline proxy/contracts/mainnet/MessageProxyForMainnet.sol]: Let's add `indexed` to this parameter and for a next event too. To make search more efficient.

File: proxy/contracts/mainnet/MessageProxyForMainnet.sol [hunk 2]
```diff
@@ -293,13 +307,14 @@ contract MessageProxyForMainnet is SkaleManagerClient, MessageProxy, IMessagePro
         require(hasRole(PAUSABLE_ROLE, msg.sender), "Incorrect sender");
         require(!pauseInfo[schainHash].paused, "Already paused");
         pauseInfo[schainHash].paused = true;
+        emit SchainPaused(schainHash);
```
Comment (payvint) [inline proxy/contracts/mainnet/MessageProxyForMainnet.sol:310]: Also I would like to add unit tests to check that events are emitted

File: proxy/contracts/mainnet/MessageProxyForMainnet.sol (hunk 1 above)
Comment (DimaStebaev) [reply proxy/contracts/mainnet/MessageProxyForMainnet.sol]: Added.

File: proxy/contracts/mainnet/MessageProxyForMainnet.sol (hunk 2 above)
Comment (DimaStebaev) [reply proxy/contracts/mainnet/MessageProxyForMainnet.sol:310]: Extended a test.

**Review by sergiy-skalelabs — APPROVED** [review]

**Review by payvint — APPROVED** [review]

**Pull request merged**